### PR TITLE
vendor: update syft to v0.66

### PR DESCRIPTION
**Pull request by jedevc**
See https://github.com/anchore/syft/releases/tag/v0.66.0 and https://github.com/anchore/syft/releases/tag/v0.66.1 (full changelist [here](https://github.com/anchore/syft/compare/v0.65.0...v0.66.1)).

### CODE REVIEW

**Review by crazy-max — APPROVED** [review]

**Comment by jedevc** [comment]
Still a bit worried by https://github.com/anchore/syft/issues/1462, probably worth waiting for a fix to land.

**Comment by crazy-max** [comment]
We should have a workflow here to ensure the image still works before merging.

**Pull request closed without merging**
